data_validation/src/spark_rapids_validation_tool/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,263 @@
+        return
+
+    result = top_level_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.f)
+    print('|--Top Level Metadata Info--|')
+    print(result.show())
+
+    result = metrics_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.pk, args.i, args.e, args.f, args.p)
+    if result.count() == 0:
+        print(f'|--Table {args.t1} and Table {args.t2} has identical metadata info--|')
+        print(result.show())
+    else:
+        print('|--Metadata Diff Info--|')
+        print(result.show())
+
+    print('|--Run Metadata Validation Success--|')
+
+def valid_input(spark, args):
+    """
+    Check the input is valida for matadata validation tool
+    1- valid table
+    2- valid included column
+    3- check format supported
+    """
+    if not valid_table(spark, args):
+        return False
+    if not valid_metadata_included_column(spark, args):
+        return False
+    if args.format != 'hive':
+        print('|--Currently only support hive format--|')
+    return True
+
+def valid_table(spark, args):
+    """
+    Check if the tables exist
+    """
+    if not spark._jsparkSession.catalog().tableExists(args.t1):
+        print(f'|--Table {args.t1} does not exist!--|')
+        return False
+    if not spark._jsparkSession.catalog().tableExists(args.t2):
+        print(f'|--Table {args.t2} does not exist!--|')
+        return False
+    return True
+
+def valid_metadata_included_column(spark, args):
+    """
+    Check if the included column valid
+    """
+    if args.i in ['None', 'all']:
+        return True
+    table_DF = load_table(spark, args.format, args.t1, args.t1p, args.pk, args.e, args.i, args.f, "")
+    excluded_columns_list = [e.strip() for e in args.e.split(",")]
+    verify_column = [i.strip() for i in args.i.split(",") if i not in excluded_columns_list]
+    verify_DF = table_DF.select(verify_column)
+
+    for c in verify_DF.schema.fields:
+        # here only excluded 'date' because it will raise exception, we also should excluded str/map/nested
+        if(any(fnmatch.fnmatch(c.dataType.simpleString(), pattern) for pattern in
+                            ['*date*'])):
+            print(f'|--Unsupported metadata included data type: {c.dataType.simpleString()} for column: {c}--|')
+            return False
+    return True
+
+def top_level_metadata(spark, format, t1, t2, t1p, t2p, f):
+    """
+    Check whether the columns number and row count could match for table1 and table2
+    """
+    if format in ['parquet', 'orc', 'csv']:
+        print('todo')
+    elif format == "hive":
+        results = []
+        table_confs = [(t1,t1p), (t2, t2p)]
+
+        for (table_name,partition) in table_confs:
+            sql = f'select * from {table_name}'
+            if any(cond != 'None' for cond in [partition, f]):
+                where_clause = ' where ' + ' and '.join(x for x in [partition, f] if x != 'None')
+            sql += where_clause
```

Comment:
This should be under `if` condition. It would throw an error if both parition and f is None because where_clause will not be set.